AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreObserveQueryTests.swift

```diff
@@ -216,119 +301,99 @@ class DataStoreObserveQueryTests: SyncEngineIntegrationTestBase {
     func testPredicateWithCreateUpdateDelete() async throws {
         await setUp(withModels: TestModelRegistration(), logLevel: .verbose)
-        try startAmplify()
+        try await startAmplifyAndWaitForReady()
         let testId = UUID().uuidString
         var snapshotCount = 0
         let predicate = Post.keys.title.beginsWith("xyz") && Post.keys.content == testId
-        let snapshotExpectation1 = expectation(description: "received snapshot 1")
-        var onSnapshot: ((DataStoreQuerySnapshot<Post>) -> Void) = { querySnapshot in
-            snapshotCount += 1
-            let items = querySnapshot.items
-            if snapshotCount == 1 {
-                self.log.info("1. \(querySnapshot)")
-                XCTAssertEqual(items.count, 0)
-                snapshotExpectation1.fulfill()
-            }
-        }
+        let snapshotExpectation1 = asyncExpectation(description: "received snapshot 1")
+        let snapshotExpectation23 = asyncExpectation(description: "received snapshot 2 / 3",
+                                                     expectedFulfillmentCount: 2)
+        let snapshotExpectation4 = asyncExpectation(description: "received snapshot 4")
+        let snapshotExpectation56 = asyncExpectation(description: "received snapshot 5 / 6",
+                                                     expectedFulfillmentCount: 2)
+        let snapshotExpectation7 = asyncExpectation(description: "received snapshot 7")
+        let snapshotExpectation8 = asyncExpectation(description: "received snapshot 8")
         Amplify.Publisher.create(Amplify.DataStore.observeQuery(for: Post.self, where: predicate)).sink { completed in
             switch completed {
             case .finished:
                 break
             case .failure(let error):
                 XCTFail("\(error)")
             }
-        } receiveValue: { onSnapshot($0) }.store(in: &cancellables)
-        await waitForExpectations(timeout: 10)
-        
-        // (1) Add model that matches predicate - should be received on the snapshot
-        let postMatchPredicate = Post(title: "xyz 1", content: testId, createdAt: .now())
-        let snapshotExpectation23 = expectation(description: "received snapshot 2 / 3")
-        snapshotExpectation23.expectedFulfillmentCount = 2
-        onSnapshot = { querySnapshot in
+        } receiveValue: { querySnapshot in
             snapshotCount += 1
             let items = querySnapshot.items
-            if snapshotCount == 2 || snapshotCount == 3 {
+            if snapshotCount == 1 {
+                self.log.info("manamaha 1. \(querySnapshot)")
+                XCTAssertEqual(items.count, 0)
+                Task { await snapshotExpectation1.fulfill() }
+            } else if snapshotCount == 2 || snapshotCount == 3 {
                 // See (1), subsequent snapshot should have item with "xyz 1".
-                self.log.info("2/3. \(querySnapshot)")
+                self.log.info("manamaha 2/3. \(querySnapshot)")
```

Comment:
let's remove `manamaha` and couple of places below. if you need a better marker, maybe use `#function` like
`self.log.info("\(#function) 2/3. ...`